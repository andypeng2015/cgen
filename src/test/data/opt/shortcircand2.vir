module foo

export function w $foo(l %a, l %i) {
@start:
  %k = slt.l %i, 3_l
  %f = flag.w %k
  %c = ne.w %f, 0_w
  br %c, @cont, @join
@cont:
  %k = sge.l %i, 0_l
  %f = flag.w %k
  jmp @join
@join:
  %c = ne.w %f, 0_w
  br %c, @ok, @bad
@ok:
  %s = mul.l %i, 4_l
  %a = add.l %a, %s
  %v = ld.w %a
  ret %v
@bad:
  ret 0xffffffff_w
}
