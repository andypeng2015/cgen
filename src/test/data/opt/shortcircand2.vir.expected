module foo

export function w $foo(l %a, l %i) {
@0:
  %1 = slt.l %i, 0x3_l ; @15
  br %1, @1, @14(0xffffffff_w)
@1:
  %6 = sge.l %i, 0x0_l ; @20
  br %6, @8, @14(0xffffffff_w)
@8:
  %4 = lsl.l %i, 0x2_l ; @18
  %5 = add.l %a, %4 ; @19
  %v.1 = ld.w %5 ; @13
  jmp @14(%v.1)
@14(%0):
  ret %0
}
